src/Utilities.hs

```diff
@@ -41,12 +47,15 @@ contextDependencies Context {..} = do
         deps <- concatMapM step pkgs
         let newPkgs = nubOrd $ sort (deps ++ pkgs)
         if pkgs == newPkgs then return pkgs else go newPkgs
-    step pkg   = case pkgCabalFile pkg of
-        Nothing        -> return [] -- Non-Cabal packages have no dependencies.
-        Just cabalFile -> do
-            deps   <- pkgDependencies cabalFile
-            active <- sort <$> stagePackages depStage
-            return $ intersectOrd (compare . pkgName) active deps
+    step pkg = pkgDependencies (ctx { Context.package = pkg }) >>= \case
+      Nothing -> return [] -- non cabal package => no dependencies
```

Comment:
Why mutilate the old comment "Non-Cabal packages have no dependencies"?

Reply:
Good question, fixed in an upcoming commit.